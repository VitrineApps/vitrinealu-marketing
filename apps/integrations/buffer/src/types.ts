import { z } from 'zod';

// Network platform types
export type Platform = 'instagram' | 'facebook' | 'twitter' | 'linkedin';
export type Subtype = 'carousel' | 'reel';

// Network constraints from config
export interface NetworkConstraints {
  maxAssets: number;
  maxFileSize: number;
  supportedFormats: string[];
  subtypes?: Record<string, {
    maxAssets: number;
    minAssets?: number;
  }>;
}

// Multi-asset update parameters
export const CreateMultiAssetUpdateParamsSchema = z.object({
  profileId: z.string(),
  text: z.string(),
  media: z.object({
    files: z.array(z.string()).min(1).max(10)
  }),
  scheduledAt: z.string().optional(),
  shortlink: z.boolean().optional(),
  utm: z.record(z.string()).optional(),
  networkHints: z.object({
    platform: z.enum(['instagram', 'facebook', 'twitter', 'linkedin']),
    subtype: z.enum(['carousel']).optional()
  }).optional()
});

export type CreateMultiAssetUpdateParams = z.infer<typeof CreateMultiAssetUpdateParamsSchema>;

// File validation result
export interface FileValidation {
  path: string;
  exists: boolean;
  mimeType?: string;
  size?: number;
  isValid: boolean;
  error?: string;
}

// Chunked update for platform constraints
export interface ChunkedUpdate {
  profileId: string;
  text: string;
  media: {
    files: string[];
  };
  scheduledAt?: string;
  shortlink?: boolean;
  utm?: Record<string, string>;
}

// Normalized response
export const MultiAssetUpdateResponseSchema = z.object({
  updateId: z.string(),
  status: z.enum(['draft', 'scheduled', 'published']),
  scheduledAt: z.string().optional(),
  chunks: z.array(z.object({
    updateId: z.string(),
    fileCount: z.number(),
    platform: z.string()
  })).optional()
});

export type MultiAssetUpdateResponse = z.infer<typeof MultiAssetUpdateResponseSchema>;

// Retry configuration
export interface RetryConfig {
  maxRetries: number;
  baseDelay: number;
  maxDelay: number;
  backoffFactor: number;
}

// Error types
export class BufferMultiAssetError extends Error {
  constructor(
    message: string,
    public code: string,
    public statusCode?: number,
    public retryable: boolean = false
  ) {
    super(message);
    this.name = 'BufferMultiAssetError';
  }
}

export class ValidationError extends BufferMultiAssetError {
  constructor(message: string, public validationErrors: FileValidation[]) {
    super(message, 'VALIDATION_ERROR', 400, false);
    this.name = 'ValidationError';
  }
}

export class PlatformLimitError extends BufferMultiAssetError {
  constructor(platform: Platform, maxAssets: number, providedAssets: number) {
    super(
      `Platform ${platform} supports maximum ${maxAssets} assets, but ${providedAssets} were provided`,
      'PLATFORM_LIMIT_EXCEEDED',
      400,
      false
    );
    this.name = 'PlatformLimitError';
  }
}