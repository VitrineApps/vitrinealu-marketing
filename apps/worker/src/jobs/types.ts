import { z } from 'zod';

export const hashJobSchema = z.object({
  fileId: z.string()
});

export const scoreJobSchema = z.object({
  fileId: z.string()
});

export const enhanceJobSchema = z.object({
  assetId: z.string().optional(),
  sourceUrl: z.string().url().optional(),
  sync: z.boolean().optional()
}).refine(data => data.assetId || data.sourceUrl, {
  message: "Either assetId or sourceUrl must be provided"
});

export const privacyBlurJobSchema = z.object({
  fileId: z.string()
});

export const backgroundCleanupJobSchema = z.object({
  fileId: z.string(),
  mode: z.enum(['clean', 'replace']).default('clean'),
  prompt: z.string().optional()
});

export const reelJobSchema = z.object({
  assetIds: z.array(z.string()).min(1),
  music: z.enum(['cinematic', 'uplift']).optional(),
  textOverlays: z.array(z.string()).optional()
});

export const linkedinJobSchema = z.object({
  assetIds: z.array(z.string()).min(1),
  narration: z.string().optional()
});

export const captionJobSchema = z.object({
  context: z.record(z.any()),
  channel: z.enum(['instagram', 'facebook', 'tiktok', 'youtube', 'linkedin'])
});

export const backgroundReplaceJobSchema = z.object({
  mediaId: z.string(),
  inputPath: z.string(),
  projectId: z.string(),
  product: z.string().optional(),
  tags: z.array(z.string()).optional(),
  preset: z.string().optional(),
  overrides: z.record(z.any()).optional(),
  maskPath: z.string().optional(),
  seed: z.number().optional(),
  callbackUrl: z.string().url().optional()
});

export type HashJobData = z.infer<typeof hashJobSchema>;
export type ScoreJobData = z.infer<typeof scoreJobSchema>;
export type EnhanceJobData = z.infer<typeof enhanceJobSchema>;
export type PrivacyBlurJobData = z.infer<typeof privacyBlurJobSchema>;
export type BackgroundCleanupJobData = z.infer<typeof backgroundCleanupJobSchema>;
export type ReelJobData = z.infer<typeof reelJobSchema>;
export type LinkedinJobData = z.infer<typeof linkedinJobSchema>;
export type CaptionJobData = z.infer<typeof captionJobSchema>;
export type BackgroundReplaceJobData = z.infer<typeof backgroundReplaceJobSchema>;

export type MediaJobData =
  | ({ kind: 'hash' } & HashJobData)
  | ({ kind: 'score' } & ScoreJobData)
  | ({ kind: 'enhance' } & EnhanceJobData)
  | ({ kind: 'privacyBlur' } & PrivacyBlurJobData)
  | ({ kind: 'backgroundCleanup' } & BackgroundCleanupJobData)
  | ({ kind: 'backgroundReplace' } & BackgroundReplaceJobData)
  | ({ kind: 'videoReel' } & ReelJobData)
  | ({ kind: 'videoLinkedin' } & LinkedinJobData)
  | ({ kind: 'caption' } & CaptionJobData);

export type MediaJobResult =
  | { kind: 'hash'; hash: string }
  | { kind: 'score'; score: number; isCandidate: boolean }
  | { kind: 'enhance'; url: string; metadata: { provider: string; scale: number; ms: number } }
  | { kind: 'privacyBlur'; url: string }
  | { kind: 'backgroundCleanup'; url: string }
  | { kind: 'backgroundReplace'; jobId: string }
  | { kind: 'videoReel'; mp4Url: string; thumbUrl: string }
  | { kind: 'videoLinkedin'; mp4Url: string; thumbUrl: string }
  | { kind: 'caption'; caption: string; hashtags: string[] }
  | { kind: 'bufferSchedule'; bufferIds: string[] };
