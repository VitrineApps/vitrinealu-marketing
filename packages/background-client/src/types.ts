/**
 * TypeScript types for the background processing client
 */

import { z } from 'zod';

// Background processing engine types
export type BGEngine = 'SDXL' | 'RUNWAY';
export type BGMode = 'transparent' | 'soften';

// Configuration schema
export const ConfigSchema = z.object({
  baseUrl: z.string().url(),
  timeout: z.number().default(300000), // 5 minutes
  retryAttempts: z.number().default(3),
  retryDelay: z.number().default(1000)
});

export type Config = z.infer<typeof ConfigSchema>;

// Cleanup request schema
export const CleanupRequestSchema = z.object({
  imagePath: z.string(),
  mode: z.enum(['transparent', 'soften']).default('transparent'),
  blurRadius: z.number().optional(),
  desaturatePct: z.number().min(0).max(100).optional(),
  enhanceFg: z.boolean().default(true),
  denoise: z.boolean().default(false)
});

export type CleanupRequest = z.infer<typeof CleanupRequestSchema>;

// Replace request schema
export const ReplaceRequestSchema = z.object({
  imagePath: z.string(),
  prompt: z.string(),
  negativePrompt: z.string().default('people, text, watermark'),
  seed: z.number().optional(),
  engine: z.enum(['SDXL', 'RUNWAY']).optional(),
  steps: z.number().min(1).max(100).default(20),
  guidanceScale: z.number().min(1).max(20).default(7.5),
  enhanceFg: z.boolean().default(true),
  matchColors: z.boolean().default(true),
  featherEdges: z.boolean().default(true)
});

export type ReplaceRequest = z.infer<typeof ReplaceRequestSchema>;

// Processing response schema
export const ProcessingResponseSchema = z.object({
  success: z.boolean(),
  outputPath: z.string().optional(),
  message: z.string().optional(),
  error: z.string().optional()
});

export type ProcessingResponse = z.infer<typeof ProcessingResponseSchema>;

// Output metadata for sidecar files
export const BackgroundMetadataSchema = z.object({
  mode: z.enum(['cleanup', 'replace']),
  engine: z.enum(['SDXL', 'RUNWAY']).optional(),
  outJpg: z.string().optional(),
  outPng: z.string().optional(),
  maskPath: z.string().optional(),
  processedAt: z.string(),
  prompt: z.string().optional(),
  settings: z.record(z.any()).optional()
});

export type BackgroundMetadata = z.infer<typeof BackgroundMetadataSchema>;

// Brand presets for automated processing
export interface BrandPreset {
  name: string;
  prompts: {
    garden: string;
    studio: string;
    minimal: string;
    lifestyle: string;
  };
  negativePrompt: string;
  settings: {
    steps: number;
    guidanceScale: number;
    engine: BGEngine;
  };
}

// Error types
export class BackgroundClientError extends Error {
  constructor(
    message: string,
    public readonly code?: string,
    public readonly statusCode?: number
  ) {
    super(message);
    this.name = 'BackgroundClientError';
  }
}

export class BackgroundTimeoutError extends BackgroundClientError {
  constructor(message: string = 'Background processing timed out') {
    super(message, 'TIMEOUT');
    this.name = 'BackgroundTimeoutError';
  }
}

export class BackgroundNetworkError extends BackgroundClientError {
  constructor(message: string, statusCode?: number) {
    super(message, 'NETWORK_ERROR', statusCode);
    this.name = 'BackgroundNetworkError';
  }
}