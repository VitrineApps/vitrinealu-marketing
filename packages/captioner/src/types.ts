import { z } from 'zod';

// Platform types
export type Platform = 'instagram' | 'tiktok' | 'youtube_shorts' | 'linkedin' | 'facebook';

export const PlatformSchema = z.enum([
  'instagram',
  'tiktok',
  'youtube_shorts',
  'linkedin',
  'facebook'
]);

// Caption JSON schema
export const CaptionJSONSchema = z.object({
  platform: PlatformSchema,
  caption: z.string().min(1, 'Caption cannot be empty'),
  hashtags: z.array(z.string()).min(1, 'At least one hashtag required'),
  call_to_action: z.string().min(1, 'Call to action cannot be empty'),
  compliance_notes: z.string().optional()
});

export type CaptionJSON = z.infer<typeof CaptionJSONSchema>;

// BrandKit type
export type BrandKit = {
  toneWords: string[];
  bannedHashtags: string[];
  locale: string;
};