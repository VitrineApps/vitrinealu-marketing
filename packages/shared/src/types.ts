export interface PipelineJob {
  id: string;
  name: string;
  payload: Record<string, unknown>;
  createdAt: string;
}

export type WebhookEventType = 'pipeline.triggered' | 'pipeline.completed' | 'approval.submitted';

export interface WebhookEvent<TPayload = unknown> {
  id: string;
  type: WebhookEventType;
  payload: TPayload;
  receivedAt: string;
}
