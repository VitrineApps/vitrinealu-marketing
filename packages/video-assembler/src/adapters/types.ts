import { z } from 'zod';
import type { AssembleJob } from '../types';

/**
 * Video adapter interface for pluggable video generation
 */
export interface IVideoAdapter {
  /**
   * Adapter name for identification
   */
  readonly name: string;

  /**
   * Generate video from assembly job
   * @param job - Video assembly job configuration
   * @returns Promise resolving to output path and metadata
   */
  generate(job: AssembleJob): Promise<VideoAdapterResult>;

  /**
   * Check if adapter is available/configured
   */
  isAvailable(): Promise<boolean>;
}

/**
 * Result from video adapter generation
 */
export interface VideoAdapterResult {
  /**
   * Path to generated video file
   */
  outPath: string;

  /**
   * Adapter that generated the video
   */
  adapter: string;

  /**
   * Generation duration in milliseconds
   */
  durationMs: number;

  /**
   * Additional metadata from the adapter
   */
  metadata?: Record<string, unknown>;
}

/**
 * CapCut API configuration
 */
export const CapCutConfigSchema = z.object({
  apiKey: z.string().min(1, 'CapCut API key is required'),
  templateId: z.string().min(1, 'CapCut template ID is required'),
  baseUrl: z.string().url().default('https://openapi-sg.capcut.com'),
  maxPollAttempts: z.number().min(1).default(60),
  pollIntervalMs: z.number().min(1000).default(5000),
});

export type CapCutConfig = z.infer<typeof CapCutConfigSchema>;

/**
 * Runway API configuration
 */
export const RunwayConfigSchema = z.object({
  apiKey: z.string().min(1, 'Runway API key is required'),
  baseUrl: z.string().url().default('https://api.runwayml.com'),
  model: z.string().default('gen-3a-turbo'),
  maxPollAttempts: z.number().min(1).default(120),
  pollIntervalMs: z.number().min(1000).default(10000),
});

export type RunwayConfig = z.infer<typeof RunwayConfigSchema>;

/**
 * Video adapter type enumeration
 */
export const VideoAdapterType = z.enum(['ffmpeg', 'capcut', 'runway']);
export type VideoAdapterType = z.infer<typeof VideoAdapterType>;

/**
 * Adapter factory configuration
 */
export interface AdapterFactoryConfig {
  defaultAdapter: VideoAdapterType;
  fallbackOrder: VideoAdapterType[];
  enableFallback: boolean;
  capcut?: CapCutConfig;
  runway?: RunwayConfig;
}

/**
 * Error types for adapter operations
 */
export class VideoAdapterError extends Error {
  constructor(
    message: string,
    public readonly adapter: string,
    public readonly code?: string,
    public readonly statusCode?: number,
  ) {
    super(message);
    this.name = 'VideoAdapterError';
  }
}

export class VideoAdapterUnavailableError extends VideoAdapterError {
  constructor(adapter: string, reason: string) {
    super(`Adapter ${adapter} is not available: ${reason}`, adapter, 'UNAVAILABLE');
    this.name = 'VideoAdapterUnavailableError';
  }
}

export class VideoAdapterTimeoutError extends VideoAdapterError {
  constructor(adapter: string, timeoutMs: number) {
    super(`Adapter ${adapter} timed out after ${timeoutMs}ms`, adapter, 'TIMEOUT');
    this.name = 'VideoAdapterTimeoutError';
  }
}