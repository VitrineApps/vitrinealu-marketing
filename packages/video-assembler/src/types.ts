import { z } from 'zod';

// Zod schemas for validation
export const CaptionLayerSchema = z.object({
  text: z.string(),
  startTime: z.number(),
  endTime: z.number(),
  fontSize: z.number().optional(),
  fontFamily: z.string().optional(),
  color: z.string().optional(),
  position: z.enum(['top', 'center', 'bottom']).optional(),
  safeArea: z.boolean().optional().default(true),
});

export const ClipSchema = z.object({
  image: z.string(),
  durationSec: z.number(),
  pan: z.enum(['none', 'leftToRight', 'rightToLeft', 'topToBottom', 'bottomToTop']).optional().default('none'),
  zoom: z.enum(['none', 'in', 'out']).optional().default('none'),
  caption: z.string().optional(),
  captionLayer: CaptionLayerSchema.optional(),
});

export const JobSchema = z.object({
  platform: z.enum(['instagram_reel', 'tiktok', 'youtube_short', 'facebook', 'linkedin']),
  clips: z.array(ClipSchema),
  outputSpec: z.object({
    width: z.number(),
    height: z.number(),
    fps: z.number().optional().default(30),
    bitrate: z.string().optional().default('2000k'),
    format: z.string().optional().default('mp4'),
  }).optional(),
  music: z.string().optional(),
  watermark: z.boolean().optional().default(true),
});

export const BrandKitSchema = z.object({
  brand: z.string(),
  colors: z.object({
    primary: z.string(),
    secondary: z.string().optional(),
    accent: z.string().optional(),
  }),
  fonts: z.object({
    primary: z.string(),
    secondary: z.string().optional(),
    heading: z.string().optional(),
  }),
  watermark: z.object({
    path: z.string(),
    opacity: z.number().optional().default(0.8),
    scale: z.number().optional().default(0.15),
    position: z.enum(['top-left', 'top-right', 'bottom-left', 'bottom-right']).optional().default('bottom-right'),
  }).optional(),
});

// TypeScript interfaces
export interface CaptionLayer {
  text: string;
  startTime: number;
  endTime: number;
  fontSize?: number;
  fontFamily?: string;
  color?: string;
  position?: 'top' | 'center' | 'bottom';
  safeArea?: boolean;
}

export interface Clip {
  image: string;
  durationSec: number;
  pan?: 'none' | 'leftToRight' | 'rightToLeft' | 'topToBottom' | 'bottomToTop';
  zoom?: 'none' | 'in' | 'out';
  caption?: string;
  captionLayer?: CaptionLayer;
}

export interface OutputSpec {
  width: number;
  height: number;
  fps?: number;
  bitrate?: string;
  format?: string;
}

export interface Job {
  platform: 'instagram_reel' | 'tiktok' | 'youtube_short' | 'facebook' | 'linkedin';
  clips: Clip[];
  outputSpec?: OutputSpec;
  music?: string;
  watermark?: boolean;
}

export interface BrandKit {
  brand: string;
  colors: {
    primary: string;
    secondary?: string;
    accent?: string;
  };
  fonts: {
    primary: string;
    secondary?: string;
    heading?: string;
  };
  watermark?: {
    path: string;
    opacity?: number;
    scale?: number;
    position?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
  };
}

export interface VideoAssemblerOptions {
  inputDir: string;
  outputPath: string;
  brandKit?: BrandKit | string;
  profile?: 'reels' | 'square' | 'landscape';
  fps?: number;
  bitrate?: string;
  watermark?: boolean;
  music?: string;
}

// Platform presets
export const PLATFORM_PRESETS: Record<string, OutputSpec> = {
  instagram_reel: { width: 1080, height: 1920, fps: 30, bitrate: '2000k' },
  tiktok: { width: 1080, height: 1920, fps: 30, bitrate: '2000k' },
  youtube_short: { width: 1080, height: 1920, fps: 30, bitrate: '2000k' },
  facebook: { width: 1280, height: 720, fps: 30, bitrate: '2500k' },
  linkedin: { width: 1920, height: 1080, fps: 30, bitrate: '3000k' },
};