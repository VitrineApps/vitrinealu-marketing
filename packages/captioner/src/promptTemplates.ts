import { Platform } from './types.js';

export interface PromptInputs {
  brief: string;
  productTags: string[];
  location?: string;
  features?: string[];
  brandTone?: string[];
  exifHints?: {
    lens: 'ultra-wide' | 'wide' | 'standard';
    timeOfDay: 'day' | 'golden_hour' | 'night';
  };
  brand?: {
    toneWords: string[];
    bannedHashtags?: string[];
    locale?: string;
  };
}

export function buildPrompt(platform: Platform, inputs: PromptInputs): string {
  const { brief, productTags, location, features, brandTone, exifHints, brand } = inputs;

  // Use brand.locale if available, else location
  const region = (brand?.locale || location) ? ` in the ${(brand?.locale || location)?.split(',')[0]} region` : '';
  const featureBullets = features ? features.map(f => `- ${f}`).join('\n') : '';
  const exifBullets = exifHints ? `- ${exifHints.lens} lens\n- taken during ${exifHints.timeOfDay}` : '';
  const toneVocab = (brand?.toneWords || brandTone) ? `Use vocabulary that is ${(brand?.toneWords || brandTone)?.join(', ')}.` : '';

  switch (platform) {
    case 'instagram':
    case 'tiktok':
      return `Create a lifestyle-focused caption for ${platform} about: ${brief}

Key benefits: ${productTags.map(tag => `benefits of ${tag}`).join(', ')}${region}

Features:
${featureBullets}
${exifBullets}

${toneVocab}

Style: Use emojis, keep 80-180 words, include 3-6 hashtags, end with CTA: "Tap link in bio for a free quote."`;

    case 'youtube_shorts':
      return `Create a YouTube Shorts caption with a concise hook first line about: ${brief}

Key benefits: ${productTags.map(tag => `benefits of ${tag}`).join(', ')}${region}

Features:
${featureBullets}
${exifBullets}

${toneVocab}

Style: Start with hook, 1-3 hashtags, end with CTA: "Learn more via our site."`;

    case 'linkedin':
      return `Create a professional LinkedIn caption about: ${brief}

Key benefits: ${productTags.map(tag => `benefits of ${tag}`).join(', ')}${region}

Features:
${featureBullets}
${exifBullets}

${toneVocab}

Style: Spec-driven, no emojis, 60-140 words, 2-4 hashtags, end with CTA: "Request a specification or site survey."`;

    case 'facebook':
      return `Create a friendly Facebook caption about: ${brief}

Key benefits: ${productTags.map(tag => `benefits of ${tag}`).join(', ')}${region}

Features:
${featureBullets}
${exifBullets}

${toneVocab}

Style: Friendly tone, 60-120 words, 3-5 hashtags.`;

    default:
      throw new Error(`Unsupported platform: ${platform}`);
  }
}