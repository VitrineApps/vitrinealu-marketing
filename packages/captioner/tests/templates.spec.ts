import { buildPrompt, PromptInputs } from '../src/promptTemplates.js';
import { Platform } from '../src/types.js';

describe('Prompt Templates', () => {
  const baseInputs: PromptInputs = {
    brief: 'Beautiful mountain landscape',
    productTags: ['nature', 'adventure'],
    location: 'Scottish Highlands',
    features: ['stunning views', 'peaceful atmosphere'],
    brandTone: ['inspiring', 'authentic'],
  };

  describe('Instagram/TikTok template', () => {
    it('should build Instagram prompt', () => {
      const result = buildPrompt('instagram', baseInputs);
      expect(result).toMatchSnapshot();
    });

    it('should build TikTok prompt', () => {
      const result = buildPrompt('tiktok', baseInputs);
      expect(result).toMatchSnapshot();
    });
  });

  describe('YouTube Shorts template', () => {
    it('should build YouTube Shorts prompt', () => {
      const result = buildPrompt('youtube_shorts', baseInputs);
      expect(result).toMatchSnapshot();
    });
  });

  describe('LinkedIn template', () => {
    it('should build LinkedIn prompt', () => {
      const result = buildPrompt('linkedin', baseInputs);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Facebook template', () => {
    it('should build Facebook prompt', () => {
      const result = buildPrompt('facebook', baseInputs);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Input variations', () => {
    it('should handle missing location', () => {
      const inputs: PromptInputs = {
        brief: 'Test brief',
        productTags: ['tag1'],
      };
      const result = buildPrompt('instagram', inputs);
      expect(result).toContain('Key benefits: benefits of tag1');
      expect(result).not.toContain('region');
    });

    it('should handle missing features', () => {
      const inputs: PromptInputs = {
        brief: 'Test brief',
        productTags: ['tag1'],
      };
      const result = buildPrompt('instagram', inputs);
      expect(result).toContain('Features:\n');
    });

    it('should handle missing brand tone', () => {
      const inputs: PromptInputs = {
        brief: 'Test brief',
        productTags: ['tag1'],
      };
      const result = buildPrompt('instagram', inputs);
      expect(result).not.toContain('vocabulary');
    });
  });
});