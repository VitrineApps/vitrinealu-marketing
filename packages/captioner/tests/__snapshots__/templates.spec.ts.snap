// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt Templates Facebook template should build Facebook prompt 1`] = `
"Create a friendly Facebook caption about: Beautiful mountain landscape

Key benefits: benefits of nature, benefits of adventure in the Scottish Highlands region

Features:
- stunning views
- peaceful atmosphere


Use vocabulary that is inspiring, authentic.

Style: Friendly tone, 60-120 words, 3-5 hashtags."
`;

exports[`Prompt Templates Instagram/TikTok template should build Instagram prompt 1`] = `
"Create a lifestyle-focused caption for instagram about: Beautiful mountain landscape

Key benefits: benefits of nature, benefits of adventure in the Scottish Highlands region

Features:
- stunning views
- peaceful atmosphere


Use vocabulary that is inspiring, authentic.

Style: Use emojis, keep 80-180 words, include 3-6 hashtags, end with CTA: "Tap link in bio for a free quote.""
`;

exports[`Prompt Templates Instagram/TikTok template should build TikTok prompt 1`] = `
"Create a lifestyle-focused caption for tiktok about: Beautiful mountain landscape

Key benefits: benefits of nature, benefits of adventure in the Scottish Highlands region

Features:
- stunning views
- peaceful atmosphere


Use vocabulary that is inspiring, authentic.

Style: Use emojis, keep 80-180 words, include 3-6 hashtags, end with CTA: "Tap link in bio for a free quote.""
`;

exports[`Prompt Templates LinkedIn template should build LinkedIn prompt 1`] = `
"Create a professional LinkedIn caption about: Beautiful mountain landscape

Key benefits: benefits of nature, benefits of adventure in the Scottish Highlands region

Features:
- stunning views
- peaceful atmosphere


Use vocabulary that is inspiring, authentic.

Style: Spec-driven, no emojis, 60-140 words, 2-4 hashtags, end with CTA: "Request a specification or site survey.""
`;

exports[`Prompt Templates YouTube Shorts template should build YouTube Shorts prompt 1`] = `
"Create a YouTube Shorts caption with a concise hook first line about: Beautiful mountain landscape

Key benefits: benefits of nature, benefits of adventure in the Scottish Highlands region

Features:
- stunning views
- peaceful atmosphere


Use vocabulary that is inspiring, authentic.

Style: Start with hook, 1-3 hashtags, end with CTA: "Learn more via our site.""
`;
